src/wasm/CMakeLists.txt

```diff
@@ -1,11 +1,22 @@
-
 faasm_private_lib(wasm
     WasmEnvironment.cpp
     WasmExecutionContext.cpp
     WasmModule.cpp
     chaining_util.cpp
 )
+
+# Populate the variables defined for cross-compiling WASM
+set(FAASM_WASM_BYTES_PER_PAGE $ENV{FAASM_WASM_BYTES_PER_PAGE})
```

Comment:
Do you have any particular reason to grab these from the environment? Usually you'd use cache variables for these sort of tunables, done with `set(... CACHE type "docstring")`, as they are less fragile if you accidentally call cmake from the wrong context

Reply:
The goal of this PR is to share this constant definitions with the CPP toolchain, and have them hard-coded only once across all the repos. IMO, even if they won't change, I'd rather have them hardcoded only once (if that doesn't add too much overhead).

So, the way I thought of sharing variables between a python script (`faasm/cpp/faasmtools/build.py`) and CMake files was through environment variables. It feels a bit hacky, so if you have another alternative I'm happy to adopt it.

Reply:
If they're in a python dict, you should be able to also just pass them in to cmake during configuration time via the `-D` flags when they're cache variables